frontend/src/tournaments/roundRobin/robinCrosstablePage.tsx

```diff
@@ -0,0 +1,143 @@
+    TableRow,
+    Typography,
+} from '@mui/material';
+import React, { useEffect, useState } from 'react';
+import {
+    cohorts,
+    fetchTournamentData,
+    fetchTournamentIds,
+    TournamentData,
+} from './roundRobinApi';
+
+const Crosstable: React.FC = () => {
+    const [selectedCohort, setSelectedCohort] = useState<number>(0);
+    const [tournamentIds, setTournamentIds] = useState<string[]>([]);
+    const [tournamentData, setTournamentData] = useState<TournamentData[]>([]);
+
+    const handleCohortChange = (event: SelectChangeEvent<number>) => {
+        setSelectedCohort(Number(event.target.value));
+    };
+
+    useEffect(() => {
+        if (selectedCohort !== 0) {
+            fetchTournamentIds(selectedCohort)
+                .then(setTournamentIds)
+                .catch(console.error);
+        }
+    }, [selectedCohort]);
+
+    useEffect(() => {
+        if (tournamentIds.length > 0) {
+            setTournamentData([]); // Clear previous tournament data
+            Promise.all(tournamentIds.map((id) => fetchTournamentData(id)))
+                .then(setTournamentData)
+                .catch(console.error);
+        }
+    }, [tournamentIds]);
+
+    useEffect(() => {
+        if (selectedCohort !== 0) {
+            fetchTournamentIds(selectedCohort);
+        }
+    }, [selectedCohort]);
+
+    useEffect(() => {
+        if (tournamentIds.length > 0) {
+            setTournamentData([]); // Clear previous tournament data
+            tournamentIds.forEach((id) => {
+                fetchTournamentData(id);
+            });
+        }
+    }, [tournamentIds]);
+
+    return (
+        <Container maxWidth='xl' sx={{ py: 5 }}>
+            <Box sx={{ display: 'flex', gap: 2, mb: 3 }}>
+                <FormControl fullWidth>
+                    <InputLabel id='cohort-selector-label'>Select Cohort</InputLabel>
+                    <Select
+                        labelId='cohort-selector-label'
+                        value={selectedCohort}
+                        onChange={handleCohortChange}
+                        label='Select Cohort'
+                    >
+                        {cohorts.map((cohort) => (
+                            <MenuItem key={cohort.value} value={cohort.value}>
+                                <CohortIcon
+                                    cohort={cohort.label}
+                                    sx={{ marginRight: '0.6em', verticalAlign: 'middle' }}
+                                    tooltip=''
+                                    size={25}
+                                />{' '}
+                                {cohort.label}
+                            </MenuItem>
+                        ))}
+                    </Select>
+                </FormControl>
+            </Box>
+
+            {tournamentData.length > 0 && (
+                <Box sx={{ mb: 3 }}>
+                    {tournamentData.map((tournament, idx) => (
+                        <div key={idx}>
```

Comment:
You could probably remove the div and just put the key on the TableContainer

Reply:
sure